include/queue.hpp

```diff
@@ -1,20 +1,42 @@
 #ifndef MY_QUEUE_QUEUE_HPP
 #define MY_QUEUE_QUEUE_HPP
 
+template<typename T>
 struct node_t {
-    int value;
+    T value;
     node_t* next;
 };
 
+template<typename T>
 class queue_t {
 public:
     queue_t() = default;
-    void enqueue(int value);
-    int dequeue();
+    void enqueue(T value);
+    T dequeue();
 
 private:
-    node_t* head = nullptr;
-    node_t* tail = nullptr;
+    node_t<T>* head = nullptr;
+    node_t<T>* tail = nullptr;
 };
 
+template<typename T>
+void queue_t<T>::enqueue(T value) {
+    auto *temp = new node_t<T>{value, nullptr};
+    if (!head) {
+        head = tail = temp;
+    } else {
+        tail->next = temp;
+        tail = tail->next;
+    }
+}
+
+template<typename T>
+T queue_t<T>::dequeue() {
+    T value = head->value;
```

Comment:
as `head->value` won't be used anymore, `T value = std::move(head->value)` can be used.